R/rbabylon.R

```diff
@@ -44,18 +44,40 @@ bbi_exec <- function(.cmd_args,
     stdout = "|",
     stderr = "2>&1"
   )
-
   if (.wait) {
     # wait for process and capture stdout and stderr
-    p$wait()
+    if (all(c("nonmem", "summary", "--json") %in% .cmd_args)) {
```

Comment:
It appears we can solve this without special handling for this command, and without introducing a timeout. The following change appears to work on both macOS and Linux:

```r
  if (.wait) {
    if (Sys.info()[["sysname"]] != "Darwin") {
      p$wait()
    }
    output <- p$read_all_output_lines()
    check_status_code(p$get_exit_status(), output, .cmd_args)
  } else {
    output <- "NO STDOUT BECAUSE `.wait = FALSE`"
  }
```

36eab35 notes that `read_all_output_lines()` is non-blocking, which would explain why `wait()` is necessary on Linux. But `wait()` actually causes problems on macOS, and `read_all_output_lines()` is sufficient. I'm not sure if that means that `read_all_output_lines()` _is_ blocking on macOS, but the behavior is definitely different.

Reply:
We agreed to

1. implement the solution in this PR rather than special handling for macOS

2. dig into the apparent `processx` issue to understand the root cause

Reply:
We are seeing a variety of errors running `check()` on Linux. It appears that increasing the `timeout`, e.g., to 20, resolves the issue, but the sensitivity to this value on different platforms is unexpected.

Reply:
We settled on 1000 ms as the safest route, though it is obviously very very generous. The thinking here is that, in the very unlikely case that a user stumbles onto this, it will still appear to hang and they will submit a bug report. If we set it to 30ms or something it a) could still fail to build and b) could come back just fast enough that a user thinks it's normal. 

For example, running 1000 model summaries for a bootstrap or something might take 2 mins when it should really take 3 seconds if things were working right. But we would never hear about it because the user might think 2 mins was slow but reasonable and _not_ submit a bug or anything.

To reiterate though: this is very much a band-aid solution. We are planning to address this more thoroughly in a future release.